### DISCUSSION TITLE
Styles absolutely broken with Safari older than 15.5

### DISCUSSION THREAD

**Discussion by messagenes** [primefaces/community#2685] [PrimeFlex] [answered]
We are using PrimeReact and PrimeFlex on our site. Some users are still running Safari versions lower than 15.5, and they are experiencing a broken UI. I would like to know if there is anything I can do to improve compatibility for those versions. Below, you will find some screenshots (left is Safari 15.0, right is Safari 15.5) with logos and brand names removed to avoid any potential advertising or spam.

---

![Screenshot at Sep 13 12-36-01](https://github.com/user-attachments/assets/ebc7f1ed-42c3-4038-ab3d-0a8f8e9edea7)

**Comment by aparajita** [comment]
Two problems:

1. PrimeReact uses `@layer`, which is only supported by Safari 15.4+.
2. The `@primeuix` themes use CSS `color-mix`, which is only supported on Safari 16.2+.

**Comment by skj-skj** [comment, marked as answer]
@messagenes you can try postcss

`postcss.config.mjs`
```javascript
export default {
  plugins: {
    'postcss-preset-env': {
      browsers: ['> 0.2%', 'not dead', 'Safari < 15.4']
    }
  }
};
```

you will need to Install `postcss-preset-env` along with `postcss`
`npm i postcss postcss-preset-env`
